soc/riscv/riscv-privilege/hi3861/soc.h

```diff
@@ -7,3 +7,15 @@
 
 #include <soc_common.h>
 #include <zephyr/devicetree.h>
+
+/* Implement the following ARM intrinsics as no-op:
+ * - ARM Data Memory Synchronization Barrier
+ * - ARM Instruction Synchronization Barrier
+ */
+#ifndef __DMB
+#define __DMB()
+#endif
+
+#ifndef __ISB
+#define __ISB()
+#endif
```

Comment:
why this is here.

Reply:
This SoC integrated a generic PL011 UART IP from ARM. The existing driver implementation calls DMB and ISB in several places. Therefore, I need these no-op defines to allow it to compile. Should I explain the purpose above the defines here, or should I mask them with something like #ifdef CONFIG_ARCH_ARM in uart_pl011.c?

https://github.com/zephyrproject-rtos/zephyr/blob/f12ffaea1a9105d0cff74ea50ecde0549c680740/drivers/serial/uart_pl011.c#L211

Reply:
move
```c
#ifndef __DMB
#define __DMB()
#endif
```
in the driver